lib/bitcoinerlab.js

```diff
@@ -0,0 +1,144 @@
+const secp256k1 = require('@exodus/bitcoinerlab-secp256k1')
+let elliptic
+
+const verbose = false
+function loadLegacy (method, message) {
+  if (verbose) {
+    if (!message) {
+      console.warn(`${method} is not implemented, using legacy approach...`)
+    } else console.warn(`${method} | ${message}`)
+  }
+  if (!elliptic) {
+    elliptic = require('./elliptic')
+  }
+}
+
+function isCompressedPublicKey (array) {
+  return array.length === 33
+}
+
+module.exports = {
+  contextRandomize: function () {
+    return 0
+  },
+  privateKeyVerify: function (privateKey) {
+    return secp256k1.isPrivate(privateKey) ? 0 : 1
+  },
+  privateKeyNegate: function (privateKey) {
+    try {
+      privateKey.set(secp256k1.privateNegate(privateKey))
+    } catch (e) {
+      loadLegacy('privateKeyNegate', e.toString())
+      elliptic.privateKeyNegate(privateKey)
+    }
+    return 0
+  },
+  privateKeyTweakAdd: function (privateKey, tweak) {
+    try {
+      privateKey.set(secp256k1.privateAdd(privateKey, tweak))
+    } catch (e) {
+      return 1
+    }
+    return 0
+  },
+  privateKeyTweakMul: function (privateKey, tweak) {
+    loadLegacy('privateKeyTweakMul')
+    return elliptic.privateKeyTweakMul(privateKey, tweak)
+  },
+  publicKeyVerify: function (publicKey) {
+    return secp256k1.isPoint(publicKey) ? 0 : 1
+  },
+  publicKeyCreate: function (output, seckey) {
+    try {
+      output.set(secp256k1.pointFromScalar(seckey, isCompressedPublicKey(output)))
+      return 0
+    } catch (e) {
+      if (e.toString().includes('Expected Private')) return 1
+      return 2
```

Comment:
I can't find this return code in elliptic, where is this from?

Reply:
Correct I also had that in mind. `index.js` does handle this error case

https://github.com/ExodusMovement/secp256k1-node/blob/ap/noble/lib/index.js#L122-L123

What would you recommend me to do?

Reply:
Maybe it's best to re-throw the error rather than claiming it's an error that it is not?

Reply:
Alright, I went ahead and `return 1` only.